### ISSUE TITLE
Implement working "hello world" Task + TaskRun

### ISSUE THREAD

**Issue by bobcatfish** [knative/build-pipeline#59]
## Expected Behavior

A user should be able to create a "hello world" Task that just echoes "hello world" and run it as many times as they like in their k8s cluster by creating TaskRuns.

Requirements:
* Test coverage (unit tests + if possible integration tests, may be blocked by #16 )
* Docs that explain how to run this Task, with examples
* Docs that explain how to debug this (i.e. where to look if it goes wrong)

## Actual Behavior

It is possible to get something like this working with Build only, but currently our TaskRun controller does nothing.

## Steps to Reproduce the Problem

  1. Deploy the Pipeline CRD + related CRDs to your cluster (note: should be docs on how to do this post kubebuilder)
  2. Create a task like this:
```yaml
apiVersion: pipeline.knative.dev/v1beta1
kind: Task
metadata:
  name: helloworld-task
  namespace: default
spec:
    buildSpec:
        steps:
            - name: helloworld
              image: busybox
              args: ['echo', 'hello world'']
```
  3. Create a task run like this:
```yaml
apiVersion: pipeline.knative.dev/v1beta1
kind: TaskRun
metadata:
  name: helloworld-task-run-0
spec:
    taskRef:
        name: helloworld-task
    trigger:
        triggerRef:
            type: manual
```
  4. Look at the logs from the running container (Note the best docs we have on this at the moment are [some of the steps in the Build quick start](https://github.com/knative/docs/blob/master/build/creating-builds.md#creating-and-running-a-build)) - these should contain "hello world"
  5. Observe that the `status` of the `TaskRun` has been updated appropriately, e.g. see [this example](https://github.com/knative/build-pipeline/blob/1c4deea8de70d40ded6cab8d64fe303d65ce2970/examples/invocations/run-kritis-test.yaml#L37) which contains:
* Outcomes of individual steps
* Started, Completed, and Successful conditions

**Comment by pivotal-nader-ziada** [comment]
in the example above, in step (3) should the name of `TaskRef` be `helloworldTask` as indicated by the name defined for the task in step (2)
or is that name coming from somewhere else?

**Comment by bobcatfish** [comment]
> in the example above, in step (3) should the name of TaskRef be helloworldTask as indicated by the name defined for the task in step (2)
> or is that name coming from somewhere else?

Oh whoops, so in step (3) I think the name should be `helloworld-task` since that's what the `Task` is called. Thanks for catching that @pivotal-nader-ziada ! I've updated it, let me know if it still doesn't seem right.

**Comment by aaron-prindle** [comment]
fixed with #86
